src/System Application/Test/Email/src/EmailTest.Codeunit.al

```diff
@@ -1412,12 +1413,14 @@ codeunit 134685 "Email Test"
 
         // [When] Retrieving emails
         // [Then] An error is thrown that the connector does not support this operation
-        asserterror Email.RetrieveEmails(EmailAccount."Account Id", EmailAccount.Connector, EmailInbox);
+        asserterror Email.RetrieveEmails(EmailAccount."Account Id", EmailAccount.Connector, EmailInbox, EmailRetrievalFilters);
         Assert.ExpectedError('The selected email connector does not support retrieving emails');
     end;
 
+#if not CLEAN26
     [Test]
-    procedure RetrieveEmails()
+    [Obsolete('v2 connector is replaced by v3 connector.', '26.0')]
```

Comment:
Same for `RetrieveEmailsFailv2()`: Why would you obsolete test functions?

Reply:
I obsoleted test functions because I want the test to stay but continue working until the interface is removed. Otherwise I'd be removing the test function now and if v2 were to fail, for any reason, we wouldn't know.

Reply:
I don't quite understand. Of course, the test functions should stay until the interface is removed. You already have put the compiler directives (#if not CLEAN26) around your v2 test functions, so they will be removed (only) together with the obsoleted interface. That's all correct, but also already sufficient. 
The obsolete attribute itself is required only to "inform" dependent extensions. I cannot imagine a scenario where anybody calls your test (!) functions in their code?

So what is it exactly you achieve with setting the Obsolete attribute here?

Reply:
Ah, hold on - does it mean you have started to use the AppSourceCop for test apps as well? Then you could not delete the public test function before obsoleting it earlier.